src/containers/Preview.js

```diff
@@ -27,7 +29,27 @@ function mapStateToProps(state) {
   };
 }
 
+function generateConsoleLogDispatcher(dispatch, timeout) {
+  let queue = [];
+
+  function flushQueue() {
+    dispatch(consoleLogBatchProduced(queue));
+    queue = [];
+  }
+
+  const timer = throttle(flushQueue, timeout, {
+    leading: true,
+    trailing: true,
+  });
+
+  return function addLogEntry(value, compiledProjectKey) {
+    queue.push({value, compiledProjectKey, key: uuid().toString()});
+    timer();
+  };
+}
+
 function mapDispatchToProps(dispatch) {
+  const logDispatcher = generateConsoleLogDispatcher(dispatch, 1000);
```

Comment:
I would still give this an imperative name like `dispatchLogEntry` etc.

Reply:
Done